### PR TITLE
Component tree collapse fix

### PR DESCRIPTION
**Pull request by mohasarc**
## Description (Optional)
Fixes the issue of selecting an element from a different component causes the component tree to collapse. (6) from the list in #1630.

To fix the issue I added a custom hook to preserve BuilderTree expanded node ids because the list is lost when the BuilderTree is rerendered. BuilderTree rerenders when the activeComponent changes.

## Related Issue(s)
Related: #1630

### CODE REVIEW

**Comment by antonmihaylov** [comment]
This seems a bit like a hack and could lead to other bugs down the line.
Would it be possible to have a better solution if we keep the expanded ids in the mobx state?

**Comment by mohasarc** [comment]
> This seems a bit like a hack and could lead to other bugs down the line. Would it be possible to have a better solution if we keep the expanded ids in the mobx state?

Yeah that make sense. The reason I didn't wanna use mobx is that it is not used anywhere else in the application.

**Comment by antonmihaylov** [comment]
True, bit we need it to persist between full rerenders so I think it's a good use case

**Review by antonmihaylov — CHANGES_REQUESTED** [review]

File: libs/frontend/modules/builder/src/hooks/useExpandedNodes.tsx [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
 import { Nullable } from '@codelab/shared/abstract/types'
-import { Key, useEffect, useState } from 'react'
+import { useEffect } from 'react'
+import { BuilderTreeModel } from '../store'
 
 export type UseExpandedNodesProps = Pick<IBuilderService, 'selectedNode'> & {
   elementTree: Nullable<IElementTree>
 }
 
+const builderTreeModel = new BuilderTreeModel({ expandedNodeIds: [] })
```
[the remark this answers is not in the document]
Comment (mohasarc) [reply libs/frontend/modules/builder/src/hooks/useExpandedNodes.tsx]: That make sense, thanks for the feedback. I implemented it as you described.

**Pull request merged**